lib/tree.rb

```diff
@@ -1,25 +1,35 @@
 class NoApplesError < StandardError; end
 
 class Tree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_reader :height, :age, :apples, :alive
 
-  def initialize
+  def initialize(height, age, apples, alive)
+    @height = height
+    @age = age 
+    @apples = apples
+    @alive = alive
   end
 
   def age!
+    add_apples unless @age < 3
+    @age += 1 
   end
 
   def add_apples
+    rand(5..15).times do @apples.push(Apple.new()) end
```

Comment:
A couple ruby style points in here:
- For blocks that fit on a single line, the preferred syntax is `{ @apples.push(Apple.new()) }`.  `do`-`end` is preferred for multi-line blocks.
- It's standard to omit the option `()` for method calls with no arguments.